### PR TITLE
Mint trackers

### PR DESCRIPTION
**Pull request by Mathieu-Be**
Adding mint counters to be displayed on the UI + `amountForMintlist` for FlatLaunchpeg

### CODE REVIEW

File: contracts/BaseLaunchpeg.sol [hunk 1]
```diff
@@ -149,6 +160,10 @@ abstract contract BaseLaunchpeg is
             revert Launchpeg__LargerCollectionSizeNeeded();
         }
 
+        if (_amountForMintlist > _collectionSize) {
+            revert Launchpeg__LargerCollectionSizeNeeded();
+        }
```
Comment (Max-3-7) [inline contracts/BaseLaunchpeg.sol:165]: Should we do that and remove the one above?

```suggestion
        if (_amountForMintlist + _amountForDevs > _collectionSize) {
            revert Launchpeg__LargerCollectionSizeNeeded();
        }
```

File: contracts/FlatLaunchpeg.sol [hunk 2]
```diff
@@ -107,7 +109,9 @@ contract FlatLaunchpeg is BaseLaunchpeg, IFlatLaunchpeg {
         }
         allowList[msg.sender] -= _quantity;
         uint256 totalCost = mintlistPrice * _quantity;
+
         _mint(msg.sender, _quantity, "", false);
+        amountMintedDuringMintlist += _quantity;
```
Comment (Max-3-7) [inline contracts/FlatLaunchpeg.sol:114]: Should we make sure in this function that it's not possible to mint more than `amountForMintlist` as we do in Launchpeg?

```solidity
if (amountMintedDuringMintlist + _quantity > amountForMintlist) {
    revert Launchpeg__MaxSupplyReached();
}
```

**Review by Max-3-7 — COMMENTED** [review]
Great stuff, thank you very much 🔥

File: contracts/BaseLaunchpeg.sol [hunk 3]
```diff
@@ -31,6 +31,9 @@ abstract contract BaseLaunchpeg is
     /// @dev It can be minted any time via `devMint`
     uint256 public override amountForDevs;
 
+    /// @notice Amount of NFTs available for the allowList mint (e.g 1000)
```
Comment (cryptofish7) [inline contracts/BaseLaunchpeg.sol:34]: `allowlist`

File: contracts/BaseLaunchpeg.sol [hunk 4]
```diff
@@ -121,6 +130,7 @@ abstract contract BaseLaunchpeg is
     /// @param _maxBatchSize Max amount of NFTs that can be minted at once
     /// @param _collectionSize The collection size (e.g 10000)
     /// @param _amountForDevs Amount of NFTs reserved for `projectOwner` (e.g 200)
+    /// @param _amountForMintlist Amount of NFTs available for the allowList mint (e.g 1000)
```
Comment (cryptofish7) [inline contracts/BaseLaunchpeg.sol:133]: `allowlist`

**Review by cryptofish7 — CHANGES_REQUESTED** [review]
Make sure you search through all instances of `allowList` and change to `allowlist`.

Likewise, do the same for `mintList` and `mintlist`.

Also, I noticed we use mintlist and allowlist synonymously, which is really confusing. Can we stick to one? (My preference would be allowlist)

**Comment by Mathieu-Be** [comment]
Changed every `mintlist` into `allowlist`. @Max-3-7 you might have some `mintlist` in the front-end as well.

**Comment by Max-3-7** [comment]
Yeah i'll update the FE so we are aligned as well @Mathieu-Be

**Pull request merged**